boards/xtensa/intel_s1000_crb/board.cmake

```diff
@@ -7,3 +7,7 @@ board_finalize_runner_args(intel_s1000
   "--ocd-jtag-instr=dsp0_gdb.txt"
   "--gdb-flash-file=load_elf.txt"
 )
+
+if(NOT "${ZEPHYR_TOOLCHAIN_VARIANT}" STREQUAL "xcc")
+  message(FATAL_ERROR "ZEPHYR_TOOLCHAIN_VARIANT != xcc. This requires xcc to build!")
```

Comment:
Out of curiosity: what is it that actually requires xcc?  I mean... I understand that no one has built a gcc toolchain for S1000 and that no one plans to, but AFAIK there are no xcc extensions in use and in theory a perfectly valid app could be generated with a gcc toolchain.  Seems like you could take this out.

Reply:
Without XCC, the #error in _soc_inthandlers.h are triggered. This is because of all the missing XCHAL_* from GCC.

Reply:
AFAIK it was never possible to build this board with anything but XCC for the S1000.

Reply:
Yeah, on Xtensa you have to have a per-SoC toolchain built from their "overlay" output that comes from the synthesis tools.  And no one has done that.  My point was that in principle this could be done, and that the platform could then be buildable with gcc.  We aren't using any xcc-specific stuff.  Pedantry, basically.

Reply:
AFAIK @@lgirdwood had a gcc toolchain based on xtools for the S1000, will check with him